rfc/mentorship-rfc.md

```diff
@@ -0,0 +1,89 @@
+
+### Align Mentors
+
+Mentors will meet together and set expectations on how to interact with mentees so that mentees have a consistent experience. 
+
+### Find Mentees
+
+Mentees will be found through self-nomination. We will put out a google form, and anyone in the community could sign up to be part of the mentorship program. From there, we will pick the same number of mentees as we have mentors. We want these mentees to be spread out between as many different companies as possible.
+
+This google form will contain some information about what areas of interest a mentee has and what area of the code they’re planning to continue working in to help us match people up.
+
+### Do Mentorship
+
+The mentorship program would last for 3-6 months (hard stop at 6 months). This timeline is somewhat flexible to account for light/heavy quarters of work. It’s really up to the judgment of the pair on whether it’s working or not. 
+
+**Mentor Expectations:** 
+
+* Detailed and prompt PR review, explaining the “why” and not just what should be done (or, loop in a subject matter expert than can provide good review)
+
+* Design guidance
+
+* Biweekly check-in, ~ 30 minutes, for general questions and help
+
+* Responsive to questions on slack
+
+**Mentee Expectations:**
+
+* Actively working on Spinnaker at a company that has a stake in Spinnaker
+
+* Responds and learns from PR review
+
+* Improve quality of the PR review they give by learning from the Mentor’s suggestions and spotting similar issues on other PRs
+
+Mentors are not measured or judged on whether or not their mentee makes it to Approver status. Instead, mentors are qualitatively evaluated on things like quality of PR feedback and  responsiveness to questions (general gauge, not automatically or robotically measured).
+
+### Recognize
+
+Being a mentor requires a time investment. We’d like to give thanks for mentors because they’ll make a huge impact. Here are some ideas on how to do that:
+
+* Can we spotlight mentors (and their companies) on our blog as permanent and visible recognition? This might help companies get excited about their engineers participating in this program because it’s recognition for them.
+
+* Can we have a list of mentors and randomly have thanks to a person pop up in an app startup banner? This is much more targeted at showing other contributors that we value mentors.
+
+* Can the TOC create some sort of “distinguished mentor” or “distinguished contributor” award which they give out for outstanding contributions?
+
+### Repeat
+
+If this is successful, we’d like to keep repeating the program. Ideally after some time (maybe 3-6 months later) past mentees are qualified and selected to be mentors. We want to set up a cycle of giving back to the community.
+
+
+## Dependencies
+
+This proposal requires someone to spend time organizing the program and providing support for mentors and mentees. Is that worth it?
```

Comment:
+1. I'd like to see 2 things defined before we decide if we want to move forward or not.

1. Owner(s). Should this be the responsibility of the TOC or SC?
2. A rough outline of what next steps are and when we should expect them to be started/done. 

This is a huge opportunity for us to grow the community and relieve pressure on some of the more active OSS maintainers (once we graduate the first round of mentees). While it might be extra work in the short term, long term we have a lot to gain by helping new contributors get their feet underneath them with the project. This project should also help inform the TOC of where knowledge gaps are and what we can do to help fill them.

Reply:
I'm really not eager to have the TOC own this. It seems like something that deserves its own SIG so that it can meet and have a publicly open forum for discussing how to improve mentorship programs. Perhaps a broad "Community Health SIG" or something to that effect? I'd like to see the community drive direction on how to best improve itself. Perhaps some TOC members are part of the SIG, but it doesn't seem appropriate to me for the TOC to out-and-out own this initiative.

Reply:
That seems fair. I'd be interested in helping drive this if we could find another lead. Maybe form SIG Contributor Experience?

Reply:
I agree with Rob, I don't think this is a TOC job. 

I'm hesitant to form a SIG with the specific purpose of this mentorship program, but it would be good to have a SIG focused  Contributor Experience! That SIG could lead mentorship, but also flesh out and potentially run with ideas for how the community could triage and respond to issues/etc. I'd be interested in joining that SIG but probably not leading it. Is there anyone in the community that you know of that's interested in this topic who would like to help lead things like this?